merge.cpp

```diff
@@ -61,6 +193,160 @@ void GetSnapshotDim(const int id, const std::string& basename, const std::string
     numSnapshots = S->numColumns();
 }
 
+void GetSnapshotTime(const int id, const std::string& basename, const std::string& varName, std::vector<double> &tSnap)
+{
+    std::string filename = basename + "/param" + std::to_string(id) + "_tSnap" + varName;
+    std::ifstream infile_tSnap(filename);
+    MFEM_VERIFY(infile_tSnap.is_open(), "Snapshot time input file does not exists.");
+
+    tSnap.clear();
+    double t = 0.0;
+    while (infile_tSnap >> t)
+    {
+        tSnap.push_back(t);
+    }
+}
+
+void GetParametricTimeWindows(const int nset, const bool rhsBasis, const std::string& basename, const int windowNumSamples, int &numBasisWindows,
+                              Array<double> &twep, std::vector<std::vector<int>> &offsetAllWindows)
+{
+    std::vector<double> tVec;
+    std::vector<std::vector<double>> tSnapX, tSnapV, tSnapE, tSnapFv, tSnapFe;
+    const int numVar = (rhsBasis) ? 5 : 3;
+    std::vector<int> numSnap(nset*numVar);
+    for (int paramID = 0; paramID < nset; ++paramID)
+    {
+        GetSnapshotTime(paramID, basename, "X", tVec);
+        reverse(tVec.begin(), tVec.end());
+        tSnapX.push_back(tVec);
+        numSnap[paramID+nset*VariableName::X] = tVec.size();
+        GetSnapshotTime(paramID, basename, "V", tVec);
+        reverse(tVec.begin(), tVec.end());
+        tSnapV.push_back(tVec);
+        numSnap[paramID+nset*VariableName::V] = tVec.size();
+        GetSnapshotTime(paramID, basename, "E", tVec);
+        reverse(tVec.begin(), tVec.end());
+        tSnapE.push_back(tVec);
+        numSnap[paramID+nset*VariableName::E] = tVec.size();
+
+        if (rhsBasis)
+        {
+            GetSnapshotTime(paramID, basename, "Fv", tVec);
+            reverse(tVec.begin(), tVec.end());
+            tSnapFv.push_back(tVec);
+            numSnap[paramID+nset*VariableName::Fv] = tVec.size();
+            GetSnapshotTime(paramID, basename, "Fe", tVec);
+            reverse(tVec.begin(), tVec.end());
+            tSnapFe.push_back(tVec);
+            numSnap[paramID+nset*VariableName::Fe] = tVec.size();
+        }
+    }
+
+    bool lastBasisWindow = false;
+    std::vector<double> tTemp(nset*numVar, 0.0);
+    std::vector<int> offsetCurrentWindow(nset*numVar, 0);
```

Comment:
See comment above, to make this a 2D vector.

Reply:
@dylan-copeland I am thinking if it is better or not to use a 2D vector for `tTemp` and `offsetCurrentWindow`. We need to find the maximum and minimum of recorded time instants. Right now we use `std::max_element` on `std::vector<double> tTemp`. I think the same does not apply to a 2D vector. On the other hand, we need to stack `offsetCurrentWindow` up to `offsetAllWindows`, which will become 3D. I think we can change the data type of `numSnap` to be a 2D vector though.